cmd/reports-controller/main.go

```diff
@@ -28,20 +31,87 @@ import (
 	kubeutils "github.com/kyverno/kyverno/pkg/utils/kube"
 	reportutils "github.com/kyverno/kyverno/pkg/utils/report"
 	apiserver "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	kubeinformers "k8s.io/client-go/informers"
 	admissionregistrationv1informers "k8s.io/client-go/informers/admissionregistration/v1"
 	metadatainformers "k8s.io/client-go/metadata/metadatainformer"
+	openreportsclient "openreports.io/pkg/client/clientset/versioned/typed/openreports.io/v1alpha1"
 	kyamlopenapi "sigs.k8s.io/kustomize/kyaml/openapi"
 )
 
-func sanityChecks(apiserverClient apiserver.Interface) error {
-	return kubeutils.CRDsInstalled(apiserverClient,
-		"clusterpolicyreports.wgpolicyk8s.io",
-		"policyreports.wgpolicyk8s.io",
+func sanityChecks(logger logr.Logger, apiserverClient apiserver.Interface, polrClient wgpolicyk8sv1alpha2.Wgpolicyk8sV1alpha2Interface, orClient openreportsclient.OpenreportsV1alpha1Interface) error {
+	crdNames := []string{
 		"ephemeralreports.reports.kyverno.io",
 		"clusterephemeralreports.reports.kyverno.io",
-	)
+	}
+	if orClient != nil {
+		crdNames = append(crdNames, "reports.openreports.io", "clusterreports.openreports.io")
+		err := kubeutils.CRDsInstalled(apiserverClient, crdNames...)
+		if err != nil {
+			return err
+		}
+
+		err = kubeutils.CRDsInstalled(apiserverClient, "clusterpolicyreports.wgpolicyk8s.io", "policyreports.wgpolicyk8s.io")
+		// there was no wgpolicy reports in the cluster, exit the function successfully
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		// error was nil, migrate wgpolicyreports to openreports
+		polrs, err := polrClient.PolicyReports(metav1.NamespaceAll).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		cpolrs, err := polrClient.ClusterPolicyReports().List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, r := range polrs.Items {
```

Comment:
why do we do this? As discussed we only delete old reports, kyverno will handle to recreate them anway. No active migration is needed.